src/protocols/radius/encode.c

```diff
@@ -735,7 +783,7 @@ static ssize_t attr_shift(uint8_t const *start, uint8_t const *end,
 	 *	"encode_value" function to take into account the header
 	 *	lengths.
 	 */
-	if ((ptr + ptr[1] + total) > end) return (ptr + ptr[1]) - start;
+	if (fr_dbuff_advance(dbuff, new_hdr_count * hdr_len) < 0) return ptr[1];
```

Comment:
returning `ptr[1]` changes the meaning of this return code.  It should return the number of bytes necessary to encode this data.

The previous code returned the total amount of data encoded since `start`.  this change just returns the total amount of data in this particular attribute.   So at the minimum, returning `ptr[1]` is different from what it was before.